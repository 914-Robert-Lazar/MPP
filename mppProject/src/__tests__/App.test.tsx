import { render} from '@testing-library/react'
import { expect, test } from 'vitest'
import App from '../App'
import AppRouter from '../router/router'


test("Renders the main page", () => {
    const rendering = render(<App/>)
    expect(rendering).toMatchSnapshot()
})

test("Rendering Add Exercise page", () => {
    const rendering2 = render(<AppRouter/>)
    expect(rendering2).toMatchSnapshot()
})
