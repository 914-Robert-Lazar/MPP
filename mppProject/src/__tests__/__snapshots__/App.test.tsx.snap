// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rendering Add Exercise page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <h2>
            Workout Exercises
          </h2>
          <div>
            <form
              class="filterForm"
            >
              <label
                for="filter"
              >
                Filter exercises by name: 
              </label>
              <input
                id="filter"
                placeholder="Search..."
                type="text"
                value=""
              />
            </form>
          </div>
          <div>
            <div>
              <table>
                <thead>
                  <tr>
                    <th>
                      Name
                    </th>
                    <th>
                      Type
                    </th>
                    <th>
                      Level
                    </th>
                    <th>
                      Actions
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Push-up
                    </td>
                    <td>
                      push
                    </td>
                    <td>
                      2
                    </td>
                    <td>
                      <button>
                        Edit
                      </button>
                      <button>
                        Delete
                      </button>
                      <button>
                        View
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Pull-up
                    </td>
                    <td>
                      pull
                    </td>
                    <td>
                      3
                    </td>
                    <td>
                      <button>
                        Edit
                      </button>
                      <button>
                        Delete
                      </button>
                      <button>
                        View
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      One leg squat
                    </td>
                    <td>
                      leg
                    </td>
                    <td>
                      4
                    </td>
                    <td>
                      <button>
                        Edit
                      </button>
                      <button>
                        Delete
                      </button>
                      <button>
                        View
                      </button>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="addButton"
            >
              <button>
                <a
                  href="/exercises/add"
                >
                  Add Exercise
                </a>
              </button>
            </div>
          </div>
          <button
            id="chartButton"
          >
            Chart
          </button>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <h2>
          Workout Exercises
        </h2>
        <div>
          <form
            class="filterForm"
          >
            <label
              for="filter"
            >
              Filter exercises by name: 
            </label>
            <input
              id="filter"
              placeholder="Search..."
              type="text"
              value=""
            />
          </form>
        </div>
        <div>
          <div>
            <table>
              <thead>
                <tr>
                  <th>
                    Name
                  </th>
                  <th>
                    Type
                  </th>
                  <th>
                    Level
                  </th>
                  <th>
                    Actions
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>
                    Push-up
                  </td>
                  <td>
                    push
                  </td>
                  <td>
                    2
                  </td>
                  <td>
                    <button>
                      Edit
                    </button>
                    <button>
                      Delete
                    </button>
                    <button>
                      View
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    Pull-up
                  </td>
                  <td>
                    pull
                  </td>
                  <td>
                    3
                  </td>
                  <td>
                    <button>
                      Edit
                    </button>
                    <button>
                      Delete
                    </button>
                    <button>
                      View
                    </button>
                  </td>
                </tr>
                <tr>
                  <td>
                    One leg squat
                  </td>
                  <td>
                    leg
                  </td>
                  <td>
                    4
                  </td>
                  <td>
                    <button>
                      Edit
                    </button>
                    <button>
                      Delete
                    </button>
                    <button>
                      View
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="addButton"
          >
            <button>
              <a
                href="/exercises/add"
              >
                Add Exercise
              </a>
            </button>
          </div>
        </div>
        <button
          id="chartButton"
        >
          Chart
        </button>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Renders the main page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div>
          <div>
            <h2>
              Workout Exercises
            </h2>
            <div>
              <form
                class="filterForm"
              >
                <label
                  for="filter"
                >
                  Filter exercises by name: 
                </label>
                <input
                  id="filter"
                  placeholder="Search..."
                  type="text"
                  value=""
                />
              </form>
            </div>
            <div>
              <div>
                <table>
                  <thead>
                    <tr>
                      <th>
                        Name
                      </th>
                      <th>
                        Type
                      </th>
                      <th>
                        Level
                      </th>
                      <th>
                        Actions
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        Push-up
                      </td>
                      <td>
                        push
                      </td>
                      <td>
                        2
                      </td>
                      <td>
                        <button>
                          Edit
                        </button>
                        <button>
                          Delete
                        </button>
                        <button>
                          View
                        </button>
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Pull-up
                      </td>
                      <td>
                        pull
                      </td>
                      <td>
                        3
                      </td>
                      <td>
                        <button>
                          Edit
                        </button>
                        <button>
                          Delete
                        </button>
                        <button>
                          View
                        </button>
                      </td>
                    </tr>
                    <tr>
                      <td>
                        One leg squat
                      </td>
                      <td>
                        leg
                      </td>
                      <td>
                        4
                      </td>
                      <td>
                        <button>
                          Edit
                        </button>
                        <button>
                          Delete
                        </button>
                        <button>
                          View
                        </button>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div
                class="addButton"
              >
                <button>
                  <a
                    href="/exercises/add"
                  >
                    Add Exercise
                  </a>
                </button>
              </div>
            </div>
            <button
              id="chartButton"
            >
              Chart
            </button>
            <div />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div>
        <div>
          <h2>
            Workout Exercises
          </h2>
          <div>
            <form
              class="filterForm"
            >
              <label
                for="filter"
              >
                Filter exercises by name: 
              </label>
              <input
                id="filter"
                placeholder="Search..."
                type="text"
                value=""
              />
            </form>
          </div>
          <div>
            <div>
              <table>
                <thead>
                  <tr>
                    <th>
                      Name
                    </th>
                    <th>
                      Type
                    </th>
                    <th>
                      Level
                    </th>
                    <th>
                      Actions
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <td>
                      Push-up
                    </td>
                    <td>
                      push
                    </td>
                    <td>
                      2
                    </td>
                    <td>
                      <button>
                        Edit
                      </button>
                      <button>
                        Delete
                      </button>
                      <button>
                        View
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      Pull-up
                    </td>
                    <td>
                      pull
                    </td>
                    <td>
                      3
                    </td>
                    <td>
                      <button>
                        Edit
                      </button>
                      <button>
                        Delete
                      </button>
                      <button>
                        View
                      </button>
                    </td>
                  </tr>
                  <tr>
                    <td>
                      One leg squat
                    </td>
                    <td>
                      leg
                    </td>
                    <td>
                      4
                    </td>
                    <td>
                      <button>
                        Edit
                      </button>
                      <button>
                        Delete
                      </button>
                      <button>
                        View
                      </button>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              class="addButton"
            >
              <button>
                <a
                  href="/exercises/add"
                >
                  Add Exercise
                </a>
              </button>
            </div>
          </div>
          <button
            id="chartButton"
          >
            Chart
          </button>
          <div />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
